### ISSUE TITLE
Wrong warning message: cannot apply unary operator `-` to type `i32`

### ISSUE THREAD

**Issue by vjjmiras** [rust-lang/rust-analyzer#23225]
As commented [here](https://github.com/rust-lang/rust-analyzer/issues/22140), since the Rust Analyzer extension on VSCode update this morning, it now warns that the unary operator - cannot be applied to signed integers and floats.

**rust-analyzer version**: (rust-analyzer version: 0.3.3025-standalone (5c156cdfb0 2026-08-23) [/Users/vicente/.vscode/extensions/rust-lang.rust-analyzer-0.3.3025-darwin-arm64/server/rust-analyzer]

**rustc version**: rustc 1.97.1 (8bab26f4f 2026-07-14) (Homebrew)

**editor or extension**: VSCode - Extension Identifier: rust-lang.rust-analyzer, Version: 0.3.3025

**code snippet to reproduce**:
```rust
fn main() {
    let x = -1;
    println!("{x}");
}
```

**Message produced**:
cannot apply unary operator `-` to type `i32`
```
[{
	"resource": "/Users/vicente/git-repos/test-unary-operator/src/main.rs",
	"owner": "rust-analyzer",
	"code": {
		"value": "E0600",
		"target": {
			"$mid": 1,
			"path": "/stable/error_codes/E0600.html",
			"scheme": "https",
			"authority": "doc.rust-lang.org"
		}
	},
	"severity": 8,
	"message": "cannot apply unary operator `-` to type `i32`",
	"source": "rust-analyzer",
	"startLineNumber": 2,
	"startColumn": 13,
	"endLineNumber": 2,
	"endColumn": 15,
	"modelVersionId": 22,
	"origin": "extHost1"
}]
```

**Comment by lnicola** [comment]
You do have `rust-src` installed, right? Does "Go to definition" work on a standard library type?

**Comment by vjjmiras** [comment]
After uninstalling 2 of the 3 instances of Rust installations, configuring the right sysroot path and a couple of VSCode restarts, it seems to be working as usual, or even better because the Go to definition feature seemed broken before.

Thanks for your patience and guidance. Still a lot to learn here :-)
